compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs

```diff
@@ -725,7 +725,8 @@ fn codegen_regular_intrinsic_call<'tcx>(
 
             // Cranelift treats stores as volatile by default
             // FIXME correctly handle unaligned_volatile_store
-            // FIXME actually do nontemporal stores if requested
+            // FIXME actually do nontemporal stores if requested (but see the LLVM backend for which
+            // architectures even allow this in a sane way)
```

Comment:
The cranelift or gcc backends could lawfully implement the stores by introducing implicit sfences that were then propagated forward by their InstCombine pass or similar, as I mentioned in the LLVM issue, like
```
[store !nt, sfence, store !nt, sfence, ..] => [store !nt, store !nt, sfence, ..]
```
I don't consider this insane. Perhaps complex? But this would both reap the benefits of these stores in actual programs and also solve the problem regarding the Rust memory model. So I would prefer we mention somewhere that there are *complex* but plausible options here even on x86, and what is important is that they *simply cannot* lower this *directly* to the nontemporal store instructions on some architectures.

Reply:
The LLVM backend could do that too, no?

Reply:
Correct, there's nothing stopping LLVM from implementing that.

Reply:
Okay so I added the comment there and made everything else reference that.